src/plugins/backend_odbc/backend_odbc.c

```diff
@@ -0,0 +1,173 @@
+/**
+ * @file
+ *
+ * @brief Source file for the ODBC backend plugin
+ *
+ * This file contains the functions that are called by the Elektra core when it uses plugins
+ *
+ * @copyright BSD License (see LICENSE.md or https://www.libelektra.org)
+ */
+
+#include "./backend_odbc.h"
+#include "./backend_odbc_get.h"
+
+#include <kdbassert.h>
+#include <kdberrors.h>
+#include <kdblogger.h>
+
+
+int ELEKTRA_PLUGIN_FUNCTION (open) (Plugin * plugin ELEKTRA_UNUSED, Key * errorKey ELEKTRA_UNUSED)
+{
+	return ELEKTRA_PLUGIN_STATUS_SUCCESS;
+}
+
+int ELEKTRA_PLUGIN_FUNCTION (init) (Plugin * plugin, KeySet * ksDefinition, Key * parentKey)
+{
+	if (!ksDefinition)
+	{
+		ELEKTRA_SET_INSTALLATION_ERRORF (parentKey, "Got NULL for the KeySet 'ksDefinition' for the mountpoint at %s\n",
+						 keyName (parentKey));
+		return ELEKTRA_PLUGIN_STATUS_ERROR;
+	}
+	else if (!plugin)
+	{
+		ELEKTRA_SET_INSTALLATION_ERRORF (parentKey, "Got NULL for the 'plugin' argument for the mountpoint at %s\n",
+						 keyName (parentKey));
+		return ELEKTRA_PLUGIN_STATUS_ERROR;
+	}
+
+	/* Parse the mountpoint definition and check if the mountpoint definition CAN be valid */
+	struct dataSourceConfig * dsConfig = fillDsStructFromDefinitionKs (ksDefinition);
+
+	if (!dsConfig)
+	{
+		ELEKTRA_SET_INSTALLATION_ERRORF (parentKey, "Could not get all necessary data from the mountpoint definition at %s\n",
+						 keyName (parentKey));
+		return ELEKTRA_PLUGIN_STATUS_ERROR;
+	}
+
+	elektraPluginSetData (plugin, dsConfig);
```

Comment:
Are you sure that this would be safe in all cases?
Is it guaranteed that this memory can be freed?
Is it sure that it was allocated with malloc or calloc and is afterwards not accessed via another pointer somewhere else.